gatling-engine/src/main/scala/io/snice/gatling/simulations/Ss7AttachmentSimulation.scala

```diff
@@ -0,0 +1,46 @@
+package io.snice.gatling.simulations
+
+import java.io.FileInputStream
+import java.nio.file.{Files, Path}
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import io.gatling.core.Predef._
+import io.snice.gatling.config.Ss7Config
+import io.snice.gatling.scenarios.Ss7BasicScenarios
+import io.snice.gatling.ss7.protocol.Ss7Protocol
+
+import scala.concurrent.duration._
+
+class Ss7AttachmentSimulation extends Simulation {
+
+  private val CONFIG_FILE_PATH = "./ss7_gatling_config.yml"
+  val ss7Config = getConfig()
+  var ss7: Ss7Protocol = Ss7Protocol(ss7Config.ss7EngineConfig)
+  val simConfig = ss7Config.ss7SimulationConfig
+
+  val interval = simConfig.intervalInMinutes.minutes
+  val ss7Scenario = Ss7BasicScenarios.ss7Attach.inject(
+    atOnceUsers(simConfig.atOnceUsers),
+    constantUsersPerSec(simConfig.startConstantUsersPerSec) during interval,
+    rampUsersPerSec(simConfig.rampRatePerSec) to simConfig.rampRateTarget during interval,
+    constantUsersPerSec(simConfig.endConstantUsersPerSec).during(interval)
+  )
+
+  setUp(ss7Scenario).protocols(ss7).maxDuration(5.minutes)
+
+  def getConfig(): Ss7Config = {
+    prepareSimulation()
+    val reader = new FileInputStream(CONFIG_FILE_PATH)
+    val mapper = new ObjectMapper(new YAMLFactory())
+    mapper.readValue(reader, classOf[Ss7Config])
+  }
+
+  def prepareSimulation(): Unit = {
+    // copy default config file from 'resources' to local directory if does not exist
+    if (!Files.exists(Path.of(CONFIG_FILE_PATH))) {
+      val reader = getClass.getResourceAsStream("/ss7_config.yml")
+      Files.copy(reader, Path.of(CONFIG_FILE_PATH))
+    }
+  }
```

Comment:
ok so i guess we have a better way to do this... We cannot just copy paste this in every simulation we want to write.

First I'd suggest to have a single Config file. We have already the `gatling.conf` and would be good to add a section there for the jss7 stuff. 

Second we need to follow and understand how Gatling config works. 
Here you get the GatlingConfiguration ([link](https://github.com/eleonoracau/snice-gatling/blob/ss7/gatling-ss7/src/main/scala/io/snice/gatling/ss7/protocol/Ss7Protocol.scala#L12)) that you should parse and pass to your protocol. 

In order to do this you can create an Ss7Config object, you can follow this example ([link](https://github.com/sniceio/snice-gatling/blob/gtp/gatling-gtp/src/main/scala/io/snice/gatling/gtp/protocol/GtpConfig.scala#L15))

And in the end in your protocol you will invoke
```
      Ss7Protocol(Ss7Config(configuration))

```

Reply:
I can try. but it seems that it is so much work just to add a field in the configuration. and we do not copy and paste every simulation. It checks if there is a config file for SS7 in the directory, if not, we copy a file to the directory, and read from that file so that when we want to change some configuration, we can just change some configuration on that file, and the run the jar again. Without having to change on the local machine, compile then scp

Reply:
Having a single configuration file is to run your application is just so much better. So I don't want to block you here, but yeah..that would be the proper way. How much work do you think it is? 

Also one question: it's not clear to me why you need to copy the file from resources to the local directory. i think you'd need to specify your resource directory in the pom.  

And all that ugly stuff to copy the file over is just to modify the jar from inside the machine? if yes, then i think you can just use this https://stackoverflow.com/questions/1224817/modifying-a-file-inside-a-jar 
so you have your file in the resource as usual and a copy of that file that you can override